File: src/test/java/io/lsdconsulting/lsd/distributed/interceptor/captor/convert/TypeConverterShould.java
package io.lsdconsulting.lsd.distributed.interceptor.captor.convert;

import feign.Response;
import org.apache.commons.io.IOUtils;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import java.io.IOException;

import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;
import static org.mockito.BDDMockito.given;

class TypeConverterShould {

    @Test
    public void convertByteArrayToString() {
        final String body = randomAlphanumeric(20);
        assertThat(TypeConverter.convert(body.getBytes()), is(body));
    }

    @Test
    public void handleNullByteArray() {
        assertThat(TypeConverter.convert((byte[])null), is(nullValue()));
    }

    @Test
    public void convertResponseBodyToString() throws IOException {
        final String body = randomAlphanumeric(20);
        final Response.Body responseBody = Mockito.mock(Response.Body.class);
        given(responseBody.asInputStream()).willReturn(IOUtils.toInputStream(body));

        assertThat(TypeConverter.convert(body.getBytes()), is(body));
    }

    @Test
    public void handleNullResponseBody() throws IOException {
        assertThat(TypeConverter.convert((Response.Body) null), is(nullValue()));
    }
}

File: src/test/kotlin/io/lsdconsulting/lsd/distributed/interceptor/captor/convert/TypeConverterShould.kt
package io.lsdconsulting.lsd.distributed.interceptor.captor.convert

import feign.Response
import io.mockk.every
import io.mockk.mockk
import org.apache.commons.io.IOUtils
import org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.charset.StandardCharsets

internal class TypeConverterShould {
    @Test
    fun `convert byte array to string`() {
        val body = randomAlphanumeric(20)
        assertThat(body.toByteArray().stringify(), `is`(body))
    }

    @Test
    @Throws(IOException::class)
    fun `convert response body to string`() {
        val body = randomAlphanumeric(20)
        val responseBody = mockk<Response.Body>()
        every { responseBody.asInputStream() } returns IOUtils.toInputStream(body, StandardCharsets.UTF_8)
        assertThat((body.toByteArray()).stringify(), `is`(body))
    }
}
